pkg/ottl/contexts/ottlspanevent/span_events.go

```diff
@@ -55,6 +57,14 @@ type Option func(*ottl.Parser[TransformContext])
 type TransformContextOption func(*TransformContext)
 
 func NewTransformContext(spanEvent ptrace.SpanEvent, span ptrace.Span, instrumentationScope pcommon.InstrumentationScope, resource pcommon.Resource, scopeSpans ptrace.ScopeSpans, resourceSpans ptrace.ResourceSpans, options ...TransformContextOption) TransformContext {
+	// calculate the event index
+	index := -1
```

Comment:
Is this default `-1` scenario possible? If so, I'd document it on the README, so users are aware that the `event_index` might hold an invalid index value.
Otherwise, I'm not pretty sure what should be done here, but only setting it to `-1` feels like ignoring a bug or something like that.

WDYT? @evan-bradley?

Reply:
I wish we could return an error. `-1` feels unsafe, I think it could lead to panics later on.  Probably safer to default to 0, but I don't like that defaulting to 0 could result in unexpected results.

I think we should implement this as a new option, which forces the caller (who already has the span event and span the event came from) to pass in the index.

Reply:
Good point, I will change the PR to make this an option instead

Reply:
with that being added as an option, do you think we should add the option to the calls of `NewTransformContext` only within `ottl` and adapt the other packages using it in a separate follow up PR?

Reply:
Ya lets separate the OTTL updates and component updates.